util/helpers_test.go

```diff
@@ -90,3 +92,48 @@ func TestMergeUniqueValues(t *testing.T) {
 		assert.True(tt, len(res) == len(a)+2)
 	})
 }
+
+func TestLDProcessor(t *testing.T) {
+	testJsonDLContextUrlStr := "http://schema.org/"
+	ldProcessor := NewLDProcessor()
+
+	t.Run("caching document loader", func(tt *testing.T) {
+		numOfLoads := 5
+		nonCachedLoader := ld.NewDefaultDocumentLoader(nil)
+		t0 := time.Now()
+		for i := 0; i < numOfLoads; i++ {
+			doc, err := nonCachedLoader.LoadDocument(testJsonDLContextUrlStr)
+			assert.NoError(tt, err)
+			assert.NotNil(tt, doc)
+		}
+		dtNonCached := time.Now().Sub(t0)
+		tt.Logf("non-cached document loader for %d tests dt: %v\n", numOfLoads, dtNonCached)
+
+		ldProcessor := NewLDProcessor()
+		t1 := time.Now()
+		for i := 0; i < numOfLoads; i++ {
+			doc, err := ldProcessor.DocumentLoader.LoadDocument(testJsonDLContextUrlStr)
+			assert.NoError(tt, err)
+			assert.NotNil(tt, doc)
+		}
+		dtCached := time.Now().Sub(t1)
+		tt.Logf("caching document loader for %d tests dt: %v\n", numOfLoads, dtCached)
+
+		assert.True(tt, dtNonCached/dtCached > 2.0)
+	})
+
+	t.Run("get context from map", func(tt *testing.T) {
+		contextMap := map[string]interface{}{
+			"dc": "http://purl.org/dc/elements/1.1/",
+			"ex": "http://example.org/vocab#",
+			"ex:contains": map[string]interface{}{
+				"@type": "@id",
+			},
+		}
+
+		activeCtx, err := ldProcessor.GetContextFromMap(contextMap)
+		//activeCtx: &{values:map[@base: processingMode:json-ld-1.1] options:0xc0001288f0 termDefinitions:map[dc:map[@id:http://purl.org/dc/elements/1.1/ @reverse:false _prefix:true] ex:map[@id:http://example.org/vocab# @reverse:false _prefix:true] ex:contains:map[@id:http://example.org/vocab#contains @reverse:false @type:@id]] inverse:map[] protected:map[] previousContext:<nil>}
```

Comment:
perhaps a note calling out that this is the expected result?

Reply:
Updated.